src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -721,6 +725,29 @@ public override void CloseAdditionalDocument(DocumentId documentId)
             CloseDocumentCore(documentId);
         }
 
+        public bool TryOpenFile(string filePath, ProjectId projectId = null, bool preview = false)
```

Comment:
Does this actually have anything to do with the VisualStudioWorkspace? If it's just a generic helper, can it be put somewhere else? Trying to avoid accumulation of any more in this type that doesn't have to be there.